src/peer_discovery.cpp

```diff
@@ -18,16 +18,23 @@
 
 #include "peer_discovery.h"
 
+#ifdef _WIN32
+#include <Ws2tcpip.h> // needed for ip_mreq definition for multicast
+#include <Windows.h>
+#else
+#include <sys/types.h>
+#endif
+
 namespace dht {
 
 constexpr char multicast_address_ipv4[10] = "224.0.0.1";
 constexpr char multicast_address_ipv6[11] = "ff12::1234";
+constexpr size_t nodeId_data_size = 20;
```

Comment:
constants in capital.
However this is not needed since you can use infohash.size() (now constexpr)